### PR TITLE
dbeaver/dbeaver#21879-[Greenplum] NPE when add foreign key and both table are empty.

### LINKED ISSUE #21879: [Greenplum] NPE when add foreign key and both table are empty
### Description

STR

1. In Greenplum 7 add foreign key, when both tables are empty
```
ava.lang.NullPointerException: Cannot invoke "org.jkiss.dbeaver.erd.model.ERDEntity.getAttributes()" because "entity" is null
	at org.jkiss.dbeaver.erd.model.ERDUtils.getAttributeByModel(ERDUtils.java:109)
	at org.jkiss.dbeaver.erd.model.ERDAssociation.resolveAttributes(ERDAssociation.java:104)
	at org.jkiss.dbeaver.erd.model.ERDAssociation.<init>(ERDAssociation.java:84)
	at org.jkiss.dbeaver.erd.ui.editor.ERDGraphicalViewer.lambda$1(ERDGraphicalViewer.java:402)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:132)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4368)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3991)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1155)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:342)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:645)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:342)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:552)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:171)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:280)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:136)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:402)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:651)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:588)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1459)
```

### DBeaver Version

23.2.5

### Operating System

mac M1 Sonoma

### Database and driver

PostgreSQL 12.12 (Greenplum Database 7.0.0 )

### Steps to reproduce

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by serjiokov**
Covered by null check.

### CODE REVIEW

**Review by E1izabeth — APPROVED** [review]

**Comment by serjiokov** [comment]
Association can be created only and only when source and target are not nulls.

**Comment by serjiokov** [comment]
image
No issues

File: plugins/org.jkiss.dbeaver.erd.model/src/org/jkiss/dbeaver/erd/model/ERDAssociation.java [hunk 1]
```diff
@@ -68,25 +68,24 @@ public ERDAssociation(ERDElement<?> sourceEntity, ERDElement<?> targetEntity, bo
...
+     * @param reflect      reflect flag
      */
-	public ERDAssociation(DBSEntityAssociation association, ERDEntity sourceEntity, ERDEntity targetEntity, boolean reflect)
-	{
-		super(association);
-		this.targetEntity = targetEntity;
-		this.sourceEntity = sourceEntity;
-
-		// Resolve association attributes
+    public ERDAssociation(DBSEntityAssociation association, @NotNull ERDEntity sourceEntity, @NotNull ERDEntity targetEntity,
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.erd.model/src/org/jkiss/dbeaver/erd/model/ERDAssociation.java]: Code style: all parameters on the same line or each parameter on its own.
Also we use `\t) {\n` as header delimiter.

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.erd.model/src/org/jkiss/dbeaver/erd/model/ERDAssociation.java (hunk 1 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.erd.model/src/org/jkiss/dbeaver/erd/model/ERDAssociation.java]: Fixed

File: plugins/org.jkiss.dbeaver.erd.model/src/org/jkiss/dbeaver/erd/model/ERDAssociation.java [hunk 2]
```diff
@@ -68,25 +68,27 @@ public ERDAssociation(ERDElement<?> sourceEntity, ERDElement<?> targetEntity, bo
...
-		super(association);
-		this.targetEntity = targetEntity;
-		this.sourceEntity = sourceEntity;
-
-		// Resolve association attributes
+    public ERDAssociation(
+        DBSEntityAssociation association,
+        @NotNull ERDEntity sourceEntity,
+        @NotNull ERDEntity targetEntity,
+        boolean reflect) {
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.erd.model/src/org/jkiss/dbeaver/erd/model/ERDAssociation.java]:
```suggestion
    public ERDAssociation(
        DBSEntityAssociation association,
        @NotNull ERDEntity sourceEntity,
        @NotNull ERDEntity targetEntity,
        boolean reflect
    ) {
```

Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.erd.model/src/org/jkiss/dbeaver/erd/model/ERDAssociation.java]: line 82 looks very strange for me.

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.erd.model/src/org/jkiss/dbeaver/erd/model/ERDAssociation.java]: but that's how we do and we discussed it several times already

**Review by E1izabeth — APPROVED** [review]

**Comment by MashaKorax** [comment]
In same tables change routing on Mikami

https://github.com/dbeaver/dbeaver/assets/84867187/850aac85-85e6-4477-973b-97effa87d976

```
java.lang.NullPointerException: Cannot read field "vNodes" because "connEdge" is null
	at org.jkiss.dbeaver.erd.ui.layout.algorithm.direct.DirectedGraphLayoutVisitor.applyConnectionResults(DirectedGraphLayoutVisitor.java:225)
	at org.jkiss.dbeaver.erd.ui.layout.algorithm.direct.DirectedGraphLayoutVisitor.applyEntityResults(DirectedGraphLayoutVisitor.java:213)
	at org.jkiss.dbeaver.erd.ui.layout.algorithm.direct.DirectedGraphLayoutVisitor.applyDiagramResults(DirectedGraphLayoutVisitor.java:184)
	at org.jkiss.dbeaver.erd.ui.layout.algorithm.direct.DirectedGraphLayoutVisitor.layoutDiagram(DirectedGraphLayoutVisitor.java:79)
	at org.jkiss.dbeaver.erd.ui.layout.GraphLayoutAuto.layout(GraphLayoutAuto.java:73)
	at org.jkiss.dbeaver.erd.ui.layout.DelegatingLayoutManager.layout(DelegatingLayoutManager.java:90)
	at org.eclipse.draw2d.Figure.layout(Figure.java:1152)
	at org.eclipse.draw2d.Figure.validate(Figure.java:1966)
	at java.base/java.util.ArrayList.forEach(Unknown Source)
	at org.eclipse.draw2d.Figure.validate(Figure.java:1967)
	at java.base/java.util.ArrayList.forEach(Unknown Source)
	at org.eclipse.draw2d.Figure.validate(Figure.java:1967)
	at java.base/java.util.ArrayList.forEach(Unknown Source)
	at org.eclipse.draw2d.Figure.validate(Figure.java:1967)
	at java.base/java.util.ArrayList.forEach(Unknown Source)
	at org.eclipse.draw2d.Figure.validate(Figure.java:1967)
	at org.eclipse.draw2d.FreeformViewport$FreeformViewportLayout.calculatePreferredSize(FreeformViewport.java:25)
	at org.eclipse.draw2d.AbstractLayout.getPreferredSize(AbstractLayout.java:102)
	at org.eclipse.draw2d.AbstractHintLayout.getPreferredSize(AbstractHintLayout.java:84)
	at org.eclipse.draw2d.Figure.getPreferredSize(Figure.java:833)
	at org.eclipse.draw2d.ScrollPaneSolver.solve(ScrollPaneSolver.java:75)
	at org.eclipse.draw2d.FigureCanvas.layoutViewport(FigureCanvas.java:302)
	at org.eclipse.draw2d.FigureCanvas$1.notifyValidating(FigureCanvas.java:270)
	at org.eclipse.draw2d.UpdateManager.fireValidating(UpdateManager.java:132)
	at org.eclipse.draw2d.DeferredUpdateManager.performValidation(DeferredUpdateManager.java:212)
	at org.eclipse.draw2d.DeferredUpdateManager.performUpdate(DeferredUpdateManager.java:187)
	at org.eclipse.draw2d.DeferredUpdateManager$UpdateRequest.run(DeferredUpdateManager.java:45)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:132)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4368)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3991)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1155)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:342)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:645)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:342)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:552)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:171)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:280)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:136)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:402)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:651)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:588)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1459)
```

**Comment by MashaKorax** [comment]
verified

**Review by serge-rider — APPROVED** [review]

**Pull request merged**